### PR TITLE
MAYH-10881 Added a more straightforward serialization interface.

### PR DESCRIPTION
**Pull request by Kheldar**

### CODE REVIEW

File: src/Arbor/File/Format/Asif/Write.hs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  liftIO $ LBS.hGetContents h
+
+-----
+
+-- This can in priciple cover all of StringZ, Text, Binary, Bitmap, and Bitstring, as well as unknown encodings.
+-- Correctly encoding the value is the responsibility of the caller.
+lazyByteStringSegment :: MonadResource m => Whatever F.Format -> (a -> LBS.ByteString) -> T.Text -> FoldM m a [Segment Handle]
+lazyByteStringSegment fmt f t = FoldM step initial extract
+  where
+    initial = genericInitial "bytestring"
```
Comment (dsturnbull) [inline src/Arbor/File/Format/Asif/Write.hs]: Can you use `t` as the filename?

File: src/Arbor/File/Format/Asif/Write.hs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+buildAsifBytestring asifType mTimestamp fld foldable = do
+  (_, _, h) <- openTempFile Nothing "asif"
+  writeAsif h asifType mTimestamp fld foldable
+  liftIO $ hSeek h AbsoluteSeek 0
+  liftIO $ LBS.hGetContents h
+
+-----
+
+-- This can in priciple cover all of StringZ, Text, Binary, Bitmap, and Bitstring, as well as unknown encodings.
+-- Correctly encoding the value is the responsibility of the caller.
```
Comment (newhoggy) [inline src/Arbor/File/Format/Asif/Write.hs]: I'm not really understanding the comment.  Can you elaborate on what the caller is supposed to do and give an example?

**Comment by AlexeyRaga** [comment]
👍

**Comment by newhoggy** [comment]
👍

**Pull request merged**